diff -cr p3scan-2.3.1/scanner_bash.c p3scan-2.3.1-1/scanner_bash.c
*** p3scan-2.3.1/scanner_bash.c	2005-11-04 00:04:37.000000000 -0500
--- p3scan-2.3.1-1/scanner_bash.c	2005-11-04 01:08:43.000000000 -0500
***************
*** 36,42 ****
  #include <sys/stat.h>
  #include <errno.h>
  #include <malloc.h>
! #include <pcre.h>
  
  #include "p3scan.h"
  #include "parsefile.h"
--- 36,42 ----
  #include <sys/stat.h>
  #include <errno.h>
  #include <malloc.h>
! #include <pcre/pcre.h>
  
  #include "p3scan.h"
  #include "parsefile.h"
diff -cr p3scan-2.3.1/scanner_basic.c p3scan-2.3.1-1/scanner_basic.c
*** p3scan-2.3.1/scanner_basic.c	2005-11-04 00:04:37.000000000 -0500
--- p3scan-2.3.1-1/scanner_basic.c	2005-11-04 01:09:02.000000000 -0500
***************
*** 36,42 ****
  #include <sys/stat.h>
  #include <errno.h>
  #include <malloc.h>
! #include <pcre.h>
  
  #include "p3scan.h"
  
--- 36,42 ----
  #include <sys/stat.h>
  #include <errno.h>
  #include <malloc.h>
! #include <pcre/pcre.h>
  
  #include "p3scan.h"
  
diff -cr p3scan-2.3.1/scanner_clamd.c p3scan-2.3.1-1/scanner_clamd.c
*** p3scan-2.3.1/scanner_clamd.c	2005-11-04 00:04:37.000000000 -0500
--- p3scan-2.3.1-1/scanner_clamd.c	2005-11-04 01:09:25.000000000 -0500
***************
*** 34,40 ****
  #include <errno.h>
  #include <sysexits.h>
  #include <stdlib.h>
! #include <pcre.h>
  #include <sys/stat.h>
  #include <fcntl.h>
  
--- 34,40 ----
  #include <errno.h>
  #include <sysexits.h>
  #include <stdlib.h>
! #include <pcre/pcre.h>
  #include <sys/stat.h>
  #include <fcntl.h>
  
